src/coreclr/jit/lower.cpp

```diff
@@ -1419,7 +1419,7 @@ GenTree* Lowering::LowerFloatArg(GenTree** pArg, CallArg* callArg)
                     break;
                 }
                 GenTree* node = use.GetNode();
-                if (varTypeIsFloating(node))
+                if (varTypeUsesFloatReg(node))
```

Comment:
Should this be `varTypeUsesFloatArgReg()`? Effectively it's no difference, except for LoongArch64.

Unrelated, but the code below seems odd:

```
                if (node->TypeGet() == TYP_DOUBLE)
                {
                    currRegNumber = REG_NEXT(REG_NEXT(currRegNumber));
                    regIndex += 2;
                }
                else
                {
                    currRegNumber = REG_NEXT(currRegNumber);
                    regIndex += 1;
                }
```
I would expect the `TYPE_DOUBLE` == 2 registers code to only apply to arm32, but it's not ifdef'ed that way.

Reply:
> I would expect the `TYPE_DOUBLE` == 2 registers code to only apply to arm32, but it's not ifdef'ed that way.

Indeed, I thought the same but decided against `#ifdef`ing it in this change to keep the scope down.

It so happens that we will never have `DOUBLE` here on ARM64 (which is the case of interest) because morph will construct the `FIELD_LIST` with `LONG`s. Looking what happens under Linux ARM soft FP, I see the same, so it seems likely that this code is actually dead.

> Should this be `varTypeUsesFloatArgReg()` ?

I'd think it's better with `varTypeUsesFloatReg`. `varTypeUsesFloatArgReg` has the meaning of "this type can be used as an argument from an FP register file", while here we're asking the question of "does this node define an FP register".